controller/controlplane/controller.go

```diff
@@ -282,20 +285,19 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			continue
 		}
 	}
-	changed := controlplane.SetDefaults(
+	_ = controlplane.SetDefaults(
 		&cp.Spec.ControlPlaneOptions,
 		defaultArgs)
-	if changed {
-		log.Debug(logger, "updating ControlPlane resource after defaults are set since resource has changed")
-		err := r.Client.Update(ctx, cp)
-		if err != nil {
-			if k8serrors.IsConflict(err) {
-				log.Debug(logger, "conflict found when updating ControlPlane resource, retrying")
-				return ctrl.Result{Requeue: true, RequeueAfter: controller.RequeueWithoutBackoff}, nil
-			}
-			return ctrl.Result{}, fmt.Errorf("failed updating ControlPlane: %w", err)
+	stop, requeue, err := extensions.ApplyExtensions(ctx, r.Client, logger, cp, r.KonnectEnabled)
+	if err != nil {
+		if !requeue {
```

Comment:
I think that the most gopher way is to have an API, when a returned error is not `nil`, ignore the rest of the returned values. IT's error-prone and rather unexpected to expect from the caller to do something about returned values when error is not `nil`

Reply:
Yep, I agree. I've done some refactoring in a127e3152cba3502c42649ab15e0c38ed3c40158. PTAL :)